### DISCUSSION TITLE
How to serialize integer into string and then deserialize back from string to integer

### DISCUSSION THREAD

**Discussion by allan-almeida1** [nlohmann/json#4350] [Q&A] [answered]
What I have is

```cpp
enum class Type {
A,
B,
C
};
```

And I want to serialize it to json but instead of using int (which is default for enum class), I want it to save to a string, so `json = Type::B` will produce "B" in the json.

Was able to do it successfully by implementing 

```cpp
void to_json(json &j, const Type &t) {
switch (t) {
case Type::A:
j = "A";
case Type::B:
.... etc
```
But when I try to deserialize it back from string to Type, I get an exception saying that "type must be integer, but is string".

Deserialization method follows same principle, so:

```cpp
void from_json(const json &j, Type &t) {
if (json == "A") {
t = Type::A;
) else if (json == "B") {
..... etc
```

I was hoping it would solve the problem of converting between int and string but it doesnt. Any help would be appreciated.

**Comment by jcracine-cimeq** [comment, marked as answer]
There's a macro you can use to serialize enums. Check [here for the documentation](https://json.nlohmann.me/features/enum_conversion/).

**Reply by allan-almeida1** [reply]
Works like a charm. Thank you!

**Reply by jcracine-cimeq** [reply]
Pleasure is mine! :)
